### PR TITLE
Add navigate command history feature

### LINKED ISSUE #134: navigate through command history
As a user, I want to navigate through my command history so that I can easily reuse a past command.

### PR DESCRIPTION
**Pull request by rama-pang**
Resolves #134.

The command history works as follows:
- Press the `Up` key to navigate to the previous command, and the `Down` key to the next command  in the history list. The cursor will be set to the end of the text. Note that if we are currently at the beginning of the history, pressing `Up` will move the cursor to the beginning of the text and not change the current text (i.e. default behavior); if we are currently at the end of the history, pressing `Down` will move the cursor to the end of the text and not change the current text (i.e. default behavior); 
- If we entered a successful command which is different from the last successful command, then the new command will be pushed into the history list. If it is the same as the last successful command, then the new command will not be pushed into the history list.
- The history list has maximum size 100. Beyond that, it will delete the least recent command history.

### CODE REVIEW

File: src/main/java/seedu/uninurse/ui/CommandBox.java [hunk 1]
```diff
@@ -82,4 +111,66 @@ public interface CommandExecutor {
...
+            currentPointer += 1;
+            return true;
+        }
+
+        /**
+         * Returns the command currently pointed to in the history list.
+         */
+        public String get() {
+            return history.get(currentPointer);
+        }
```
Comment (BlopApple) [inline src/main/java/seedu/uninurse/ui/CommandBox.java:173]: I think you should let `handleUpKey` and `handleDownKey` return the current string instead of using the `get` method as now you are making an impure boolean method since it is modifying an internal variable and returning the status of the modification. 

Since `CommandHistoryList` is meant to return the command at the current pointer, the logic should be handled within itself when invoked by `handleUpKey` and `handleDownKey` respectively, rather than modifying an internal variable and verifying the status by an external agent (in this case it would be the `CommandBox` object) and retrieving another internal variable as now the logic is handled by an external agent. This introduces unnecessary coupling between these 2 classes (even though functionally it is not a big issue as `CommandHistoryList` is an internal class of `CommandBox`).

In summary, I think it would be better if you could update the implementation :D

**Review by BlopApple — CHANGES_REQUESTED** [review]
Looks ok for the most part except for a minor issue on the implementation of `CommandHistoryList`. I hope you can consider some of the suggestions I gave :D

File: src/main/java/seedu/uninurse/ui/CommandBox.java [hunk 2]
```diff
@@ -82,4 +111,66 @@ public interface CommandExecutor {
...
+        private final List<String> history;
+        private int currentPointer;
+
+        /**
+         * Creates a {@code CommandHistoryList}.
+         */
+        public CommandHistoryList() {
+            history = new ArrayList<String>();
+            history.add("");
+            currentPointer = 0;
```
[the remark this answers is not in the document]
Comment (rama-pang) [reply src/main/java/seedu/uninurse/ui/CommandBox.java]: Done
